### PR TITLE
Adds router & resource.

### PR DESCRIPTION
**Pull request by alanpeabody**
Router DSL example: https://github.com/AgilionApps/elixir_blog_api/blob/feda3f3c934661cceb8ae955aa07fe7cbb9c6358/lib/blog/api.ex

Resource DSL example: https://github.com/AgilionApps/elixir_blog_api/blob/feda3f3c934661cceb8ae955aa07fe7cbb9c6358/lib/blog/api/v1/posts.ex

### CODE REVIEW

File: lib/json_api/README.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 A JsonAPI.org implimentation in Elixir.
 
+## Rational
+
+TLDR: Generate a proper JsonAPI.org server as simply as possible.
+
+In order to drive Elixir adoption in my day to day work (Agilion.com) the absolute first thing needed is an easy way to build JSON Rest APIs suitable for comsumption by Ember.js. While this is certainly acheivable with existing Elixir tooling (like the powerful Pheonix Framework), I was looking for something that gave me exactly what was needed and nothing that wasn't.
```
Comment (beerlington) [inline lib/json_api/README.md]: _achievable_

**Comment by beerlington** [comment]
:+1: I like the `find_all`, `find_many`, `find_one` pattern. Seems a lot better than trying to spread that logic across just two controller actions in Rails. How would this work if you needed to scope the queries to a specific user? Can you have the concept of a "current_user" here?

File: lib/json_api/README.md [hunk 2]
```diff
@@ -2,6 +2,15 @@
 
 A JsonAPI.org implimentation in Elixir.
 
+## Rational
```
Comment (patrickberkeley) [inline lib/json_api/README.md]: rationale

File: lib/json_api/README.md (hunk 1 above)
Comment (alanpeabody) [inline lib/json_api/README.md]: Yeah, lots of README issues. Need to update the whole thing to reflect what I actually came up with.

**Comment by alanpeabody** [comment]
@beerlington I actually stole the find_all/find_many/find_one from the Go implementation. 

There isn't anything built in for current_user. However "plugs" are actually the call stack of the request.

You could set the current user in the router:

``` elixir
plug :set_current_user
plug :match
plug :dispatch

def set_current_user(conn, _opts) do
  case User.authenticate(conn.headers) do
    nil ->    conn
    user -> Plug.Conn.assign(conn, :current_user, user)
  end
end
```

And then in the resource require the current user:

``` elixir

plug :require_current_user, only: [:create]
plug :match
plug :dispatch

def require_current_user(conn, opts) do
   #conditional
   Plug.Conn.halt(conn)
end
```

And then in the action:

``` elixir

def find_many(conn, ids) do
  okay(conn, Comment.find_by_user(conn.assigns[:current_user]))
end
```

File: lib/json_api/README.md (hunk 2 above)
Comment (alanpeabody) [inline lib/json_api/README.md]: thx

**Pull request merged**